frontend/model/captureLogs.js

```diff
@@ -0,0 +1,186 @@
+import sbp from '~/shared/sbp.js'
+import { CAPTURED_LOGS, SET_APP_LOGS_FILTER } from '~/frontend/utils/events.js'
+
+/*
+  giConsole/count - total logs stored
+  giConsole/lastEntry - latest log hash.
+  giConsole/markers - an array of markers at every 1000th entry.
+*/
+const ENTRIES_LIMIT = 4000
+const ENTRIES_MARKER = 1000
+
+let isCapturing = true
+let isConsoleOveride = false
+let lastEntry = null
```

Comment:
The `lastEntry` value must be retrieved from `localStorage`. If it doesn't exist, then it should be set to `null`. I do see how you've done it, but the way it's currently implemented has the following bug: it only works the first time. On subsequent page reloads or visits, it will be reset to `null`, when instead it should be retrieving it from `localStorage`.